lib/auth/auth_with_roles.go

```diff
@@ -239,26 +244,39 @@ func hasLocalUserRole(checker services.AccessChecker) bool {
 	return ok
 }
 
+var sessionTrackerRoles = []types.SystemRole{types.RoleKube, types.RoleNode, types.RoleAuth, types.RoleProxy, types.RoleDatabase, types.RoleApp, types.RoleWindowsDesktop}
+
+const sessionTrackerAccessDeniedMessage = "this request can only be executed by a node, proxy, kube, db, app, or windows desktop service"
+
 // CreateSessionTracker creates a tracker resource for an active session.
-func (a *ServerWithRoles) CreateSessionTracker(ctx context.Context, req *proto.CreateSessionTrackerRequest) (types.SessionTracker, error) {
-	if !a.hasBuiltinRole(string(types.RoleKube)) && !a.hasBuiltinRole(string(types.RoleNode)) && !a.hasBuiltinRole(string(types.RoleProxy)) {
-		return nil, trace.AccessDenied("this request can be only executed by a node, proxy or kube service")
+func (a *ServerWithRoles) CreateSessionTracker(ctx context.Context, tracker types.SessionTracker) (types.SessionTracker, error) {
+	if !a.hasBuiltinRole(sessionTrackerRoles...) {
+		return nil, trace.AccessDenied(sessionTrackerAccessDeniedMessage)
 	}
 
-	return a.authServer.CreateSessionTracker(ctx, req)
+	tracker, err := a.authServer.CreateSessionTracker(ctx, tracker)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return tracker, nil
+
 }
 
 // GetSessionTracker returns the current state of a session tracker for an active session.
 func (a *ServerWithRoles) GetSessionTracker(ctx context.Context, sessionID string) (types.SessionTracker, error) {
-	if !a.hasBuiltinRole(string(types.RoleKube)) && !a.hasBuiltinRole(string(types.RoleNode)) && !a.hasBuiltinRole(string(types.RoleProxy)) {
-		return nil, trace.AccessDenied("this request can be only executed by a node, proxy or kube service")
+	if !a.hasBuiltinRole(sessionTrackerRoles...) {
+		return nil, trace.AccessDenied(sessionTrackerAccessDeniedMessage)
 	}
 
 	return a.authServer.GetSessionTracker(ctx, sessionID)
 }
 
 // GetActiveSessionTrackers returns a list of active session trackers.
 func (a *ServerWithRoles) GetActiveSessionTrackers(ctx context.Context) ([]types.SessionTracker, error) {
+	if !a.hasBuiltinRole(sessionTrackerRoles...) {
```

Comment:
Is this strictly necessary?

I agree with builtin role checks for _creating_ session trackers, but maybe there are uses where plugins or other API consumers want to get info about active sessions. Should we allow this?

Reply:
Sure, I'll update `GetActiveSessionTrackers` and `GetSessionTracker` to instead check for `read`, `list` on `session_tracker`.